server/routes/users.js

```
@@ -15,33 +15,61 @@ app.get('/', authenticate, async(req, res) => {
 });
 
 app.post('/', authenticate, async(req, res) => {
-
     try {
-        let encrypted = encrypt(req.body.password);
-        //validate email is needed
-
-        if (encrypted == null) {
+        let encrypted = "";
+        let emailExist = false;
+        if(req.body.password !== "" && req.body.password !== undefined && req.body.password !== null){
```

All this validation should have been done first and fail the invokation if any required parameter is missing.